lib/src/manager_dio.dart

```diff
@@ -38,132 +38,133 @@ class DioCacheManager {
-      return options;
+      return handler.next(options);
     }
     var responseDataFromCache = await _pullFromCacheBeforeMaxAge(options);
     if (null != responseDataFromCache) {
-      return _buildResponse(
-          responseDataFromCache, responseDataFromCache?.statusCode, options);
+      return handler.resolve(_buildResponse(
+          responseDataFromCache, responseDataFromCache.statusCode, options), true);
     }
-    return options;
+    return handler.next(options);
   }
 
-  _onResponse(Response response) async {
-    if ((response.request.extra[DIO_CACHE_KEY_TRY_CACHE] ?? false) == true &&
-        response.statusCode >= 200 &&
-        response.statusCode < 300) {
+  _onResponse(Response response, ResponseInterceptorHandler handler) async {
+    if ((response.requestOptions.extra[DIO_CACHE_KEY_TRY_CACHE] ?? false) == true &&
+        response.statusCode != null &&
+        response.statusCode! >= 200 &&
+        response.statusCode! < 300) {
       await _pushToCache(response);
     }
-    return response;
+    return handler.next(response);
   }
 
-  _onError(DioError e) async {
-    if ((e.request.extra[DIO_CACHE_KEY_TRY_CACHE] ?? false) == true) {
-      var responseDataFromCache = await _pullFromCacheBeforeMaxStale(e.request);
+  _onError(DioError e, ErrorInterceptorHandler handler) async {
+    if ((e.requestOptions.extra[DIO_CACHE_KEY_TRY_CACHE] ?? false) == true) {
+      var responseDataFromCache = await _pullFromCacheBeforeMaxStale(e.requestOptions);
       if (null != responseDataFromCache)
         return _buildResponse(responseDataFromCache,
-            responseDataFromCache?.statusCode, e.request);
+            responseDataFromCache.statusCode, e.requestOptions);
     }
-    return e;
+    return handler.next(e);
   }
 
   Response _buildResponse(
-      CacheObj obj, int statusCode, RequestOptions options) {
-    Headers headers;
+      CacheObj obj, int? statusCode, RequestOptions options) {
+    Headers? headers;
     if (null != obj.headers) {
       headers = Headers.fromMap((Map<String, List<dynamic>>.from(
-              jsonDecode(utf8.decode(obj.headers))))
+              jsonDecode(utf8.decode(obj.headers!))))
           .map((k, v) => MapEntry(k, List<String>.from(v))));
     }
     if (null == headers) {
       headers = Headers();
-      options.headers.forEach((k, v) => headers.add(k, v ?? ""));
+      options.headers.forEach((k, v) => headers!.add(k, v ?? ""));
     }
     // add flag
     headers.add(DIO_CACHE_HEADER_KEY_DATA_SOURCE, "from_cache");
     dynamic data = obj.content;
     if (options.responseType != ResponseType.bytes) {
       data = jsonDecode(utf8.decode(data));
     }
+    final requestOptions = options;
+    requestOptions.extra..remove(DIO_CACHE_KEY_TRY_CACHE);
     return Response(
         data: data,
         headers: headers,
-        extra: options.extra..remove(DIO_CACHE_KEY_TRY_CACHE),
+        requestOptions: requestOptions,
```

Comment:
Could this just be
```
requestOptions: options.extra..remove(DIO_CACHE_KEY_TRY_CACHE);
```
Is there any need to reassign options to requestOptions variable?

Reply:
`options.extra.remove(DIO_CACHE_KEY_TRY_CACHE)` would return the removed object so that is not what you want to assign to `requestOptions`. I think this one-liner would work though
```
requestOptions: options.copyWith(extra: options.extra..remove(DIO_CACHE_KEY_TRY_CACHE)),
```
through it does the same thing.

Reply:
I see, Thanks!